arviz/tests/test_diagnostics.py

```diff
@@ -1,48 +1,46 @@
 import numpy as np
-import pandas as pd
 from numpy.testing import assert_allclose
-from ..stats import gelman_rubin, effective_n, geweke
-
-good_rhat = 1.1
-
-
-def fake_trace(n_samples):
-    """
-    Creates a fake trace with 3 variables and 2 chains
-    """
-    a = np.repeat((1, 5, 10), n_samples)
-    b = np.repeat((1, 5, 1), n_samples)
-    data = np.random.beta(a, b).reshape(-1, n_samples//2)
-    trace = pd.DataFrame(data.T, columns=['a', 'a', 'b', 'b', 'c', 'c'])
-    return trace
-
-
-def test_gelman_rubin():
-    """Confirm Gelman-Rubin statistic is close to 1 for a large number of samples.
-    Also checks the correct shape"""
-    trace = fake_trace(1000)
-    rhat = gelman_rubin(trace)
-    assert all(1 / good_rhat < r < good_rhat for r in rhat.values)
-    assert rhat.shape == (3,)
-
-
-def test_gelman_rubin_bad():
-    """Confirm Gelman-Rubin statistic is far from 1 for a small number of samples."""
-    trace = fake_trace(6)
-    rhat = gelman_rubin(trace)
-    assert not all(1 / good_rhat < r < good_rhat for r in rhat.values)
-
+import pandas as pd
 
-def test_effective_n():
-    n_samples = 1000
-    trace = fake_trace(n_samples)
-    eff_n = effective_n(trace)
-    assert_allclose(eff_n, n_samples, 2)
-    assert eff_n.shape == (3,)
+from ..stats import gelman_rubin, effective_n, geweke
 
 
-def test_geweke():
-    trace = fake_trace(1000)
-    gw = geweke(trace)
-    assert max(abs(gw['a'][:, 1])) < 1
-    assert max(abs(gw['a'][:, 0])) > -1
+class TestDiagnostics(object):
```

Comment:
What is the advantage of using classes for tests?

Reply:
I did this mostly to:

1. stop code from running in the global namespace:  all that code actually runs while `pytest` is _discovering_ tests, so `pytest -x` would run all the stuff outside of functions, then start running the tests and stop on the first failure.  Now it will start running the tests, only doing setup it needs to do.
2. create more granular tests: so failures would be easier to spot
3. convention: it is convention to have `test_foo.py` to test `foo.py`, and then `class TestBar` to test the `Bar` class in `foo.py`.  This makes finding where a function/method/class/module is easier.